californium-core/src/test/java/org/eclipse/californium/core/test/ClientAsynchronousTest.java

```diff
@@ -157,6 +157,46 @@ public void testAsyncObserveTriggersOnLoad() throws Exception {
 		Thread.sleep(50);
 		assertThat(receivedNotifications.get(), is(3));
 	}
+	
+	/**
+	 * Test case for CoapClient.observe(Request request, CoapHandler handler).
+	 * 
+	 * @author Praful Bhatnagar (prafulbhatnagar@gmail.com)
+	 * @throws Exception
+	 */
+	@Test
+	public void testAsyncObserveWithRequestTriggersOnLoad() throws Exception {
```

Comment:
see my comment for the other test method you added. I think we can refrain to simply make sure to detect a missing *observe* option in the passed-in `Request` ...